### PR TITLE
Fix change location.href to open tab by url in block documentation

### PR DESCRIPTION
**Pull request by tavriaforever**
- Fix support browser`s `back-next` button for tabs
- Fix `clear location.hash` after open another tab pane
- Support browser`s like IE8, that not supported `html5 history` API, they have fallback to hashbang `#!/tabname`
- Add bem-history library as bower dependencies
- Now the tab stay link to support e-readers
- Refactor code and add `jsdoc`

// cc @tadatuta @gela-d

### CODE REVIEW

File: src/blocks/common.blocks/block/block.js [hunk 1]
```diff
@@ -1,110 +1,222 @@
...
+        /**
+         * Get current current tab`s name from location.href
+         * Example 1: https://ru.bem.info/libs/bem-core/button/jsdoc/ -> jsdoc
+         * Example 2(old browser): https://ru.bem.info/libs/bem-core/button/#!/jsdoc/ -> jsdoc
+         * Example 3: https://ru.bem.info/libs/bem-core/button/ -> null
+         * @returns {String|null} - tab name
+         * @private
+         */
+        _getTabNameByPath: function () {
+            var name = window.location.href.match(/\/(docs|jsdoc|examples)/),
```
Comment (tadatuta) [inline src/blocks/common.blocks/block/block.js]: cache `window.location`

**Comment by tadatuta** [comment]
looks fine if it works

File: src/blocks/common.blocks/block/block.js [hunk 2]
```diff
@@ -1,110 +1,222 @@
...
+                tabName = $tab.attr('href');
+
+            _this._setUrl(tabName);
+            _this._loadExamples($tab);
+        },
+
+        /**
+         * Handler on click back/next button in browser
+         * open the tab by changed url in address bar
+         * For browsers, that not support history API,
```
Comment (gela-d) [inline src/blocks/common.blocks/block/block.js]: For -> for

File: src/blocks/common.blocks/block/block.js [hunk 3]
```diff
@@ -1,110 +1,222 @@
...
+         * bem-history generate `popstate` event on every hashchange for support back/next buttons,
+         * but in our cases this brings us some problem:
```
Comment (gela-d) [inline src/blocks/common.blocks/block/block.js]: this -> it

File: src/blocks/common.blocks/block/block.js [hunk 4]
```diff
@@ -1,110 +1,222 @@
...
+
+            var state = this._history.state,
+                tabName = state && (state.tab || state.data && state.data.tab);
+
+            this._openTab(tabName);
+        },
+
+        /**
+         * Open tab pane first, when page was loaded
+         * If location.pathname contain tabName, this tab will be activate
```
Comment (gela-d) [inline src/blocks/common.blocks/block/block.js]: activate -> activated

File: src/blocks/common.blocks/block/block.js [hunk 5]
```diff
@@ -1,110 +1,222 @@
...
+            var $tab = tabName ? this._getTabByName(tabName) : this._tabs.getTab(0);
+
+            this._tabs.setActiveTab($tab);
+            this._loadExamples($tab);
+        },
+
+        /**
+         * Get jQuery tab elem by passed name
+         * @param {String} tabName - name of the tab, that we want to get
+         * @returns {jQuery|null} - if we not found the tab, return null
```
Comment (gela-d) [inline src/blocks/common.blocks/block/block.js]: maybe `if we didn't found tab`?

File: src/blocks/common.blocks/block/block.js [hunk 6]
```diff
@@ -1,110 +1,222 @@
...
-            if (example.hasMod('view', 'default') && _this._getTabName() !== 'examples' && !window.history) {
-                return false;
+            return tabName;
+        },
+
+        /**
+         * Special method to get original path without tab name
+         * This is needed to fix duplicating the name of the tab in the browser`s address bar,
+         * e.g 'button/docs/jsdoc/examples/docs/'.
+         * When the name of the tab has set already, get path without name of the tab
```
Comment (gela-d) [inline src/blocks/common.blocks/block/block.js]: `has set already` -> `has already set`

File: src/blocks/common.blocks/block/block.js [hunk 7]
```diff
@@ -1,110 +1,222 @@
...
+         * @private
+         */
+        _setUrl: function (tabName) {
+            this._history.changeState('push', { tab: tabName, title: '', url: this._getUrl(tabName) });
+        },
+
+        /**
+         *  Lazy loading for default examples in tab 'examples'
+         *  We found all example inside block and check if this example not inline
+         *  and in this time example not active - do not load this example
```
Comment (gela-d) [inline src/blocks/common.blocks/block/block.js]: and if example isn't activated

File: src/blocks/common.blocks/block/block.js (hunk 3 above)
Comment (tavriaforever) [inline src/blocks/common.blocks/block/block.js]: ok

File: src/blocks/common.blocks/block/block.js (hunk 4 above)
Comment (tavriaforever) [inline src/blocks/common.blocks/block/block.js]: ok

File: src/blocks/common.blocks/block/block.js (hunk 5 above)
Comment (tavriaforever) [inline src/blocks/common.blocks/block/block.js]: yes, you are right

File: src/blocks/common.blocks/block/block.js (hunk 6 above)
Comment (tavriaforever) [inline src/blocks/common.blocks/block/block.js]: ok

File: src/blocks/common.blocks/block/block.js (hunk 7 above)
Comment (tavriaforever) [inline src/blocks/common.blocks/block/block.js]: ok

**Comment by tavriaforever** [comment]
Update PR, fix comments, rebased

**Pull request merged**